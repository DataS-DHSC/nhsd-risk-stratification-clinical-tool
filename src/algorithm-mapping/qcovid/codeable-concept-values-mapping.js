export const SNOMED_CODE_NONE = '260413007';
export const QCOVID_CODE_NONE = 'NONE';

/**
 * This file includes information and util functions needed to convert form fields
 * values for observations of type CODEABLE_CONCEPT
 */

export const VALUE_OBSERVATIONS = {
  // ethnicity
  bangladeshi: { label: 'Bangladeshi', snomedCode: '186003008' },
  chinese: { label: 'Chinese', snomedCode: '33897005' },
  indian: { label: 'Indian', snomedCode: '110751000000108' },
  pakistani: { label: 'Pakistani', snomedCode: '92461000000105' },
  'other-asian': {
    label: 'Another Asian background',
    snomedCode: '92701000000102',
  },
  african: { label: 'African', snomedCode: '92491000000104' },
  caribbean: { label: 'Caribbean', snomedCode: '107691000000105' },
  'other-black': {
    label: 'Another Black background',
    snomedCode: '92501000000105',
  },
  'asian-white': { label: 'Asian and White', snomedCode: '92441000000109' },
  'african-white': {
    label: 'Black African and White',
    snomedCode: '92431000000100',
  },
  'caribbean-white': {
    label: 'Black Caribbean and White',
    snomedCode: '92421000000102',
  },
  'other-mixed': {
    label: 'Another Mixed background',
    snomedCode: '92451000000107',
  },
  british: {
    label: 'British, English, Northern Irish, Scottish, or Welsh',
    snomedCode: '494131000000105',
  },
  irish: { label: 'Irish', snomedCode: '494161000000100' },
  'other-white': {
    label: 'Another White background',
    snomedCode: '92411000000108',
  },
  'other-ethnic-group': {
    label: 'Another ethnic background',
    snomedCode: '94151000000105',
  },
  'not-saying': { label: 'Prefer not to say', snomedCode: '92531000000104' },

  // sex
  female: { label: 'Female', snomedCode: '248152002' },
  male: { label: 'Male', snomedCode: '248153007' },

  // residential
  'not-homeless-or-resident-of-care-home': {
    label: 'Not homeless or resident of a care home',
    snomedCode: '160724009',
  },
  homeless: { label: 'Homeless', snomedCode: '266935003' },
  'resident-of-a-care-home': {
    label: 'Resident of a care home',
    snomedCode: '248171000000108',
  },

  // diabetes
  'has-no-diabetes': { label: 'Has no diabetes', snomedCode: QCOVID_CODE_NONE },
  'has-type-1-diabetes': { label: 'Has type 1', snomedCode: 'TYPE_1_DIABETES' },
  'has-type-2-diabetes': { label: 'Has type 2', snomedCode: 'TYPE_2_DIABETES' },

  // kidney, renal
  'no-cdk-or-has-ckd-1-or-2': {
    label: 'No CKD, or has CKD 1 or 2',
    snomedCode: QCOVID_CODE_NONE,
  },
  'has-cdk3': { label: 'Has CKD3', snomedCode: 'CKD3' },
  'has-cdk4': { label: 'Has CKD4', snomedCode: 'CKD4' },
  'has-cdk5-without-dialysis-or-transplant': {
    label: 'Has CKD5 without dialysis or transplant',
    snomedCode: 'CKD5_NO_DIALYSIS',
  },
  'has-cdk5-with-dialysis-in-the-last-12-months': {
    label: 'Has CKD5 with dialysis in the last 12 months',
    snomedCode: 'CKD5_DIALYSIS',
  },
  'has-cdk5-with-transplant': {
    label: 'Has CKD5 with transplant',
    snomedCode: 'CKD_TRANSPLANT',
  },

  // learning disability
  'has-no-learning-disability-or-downs-syndrome': {
    label: "Has no learning disability or Down's syndrome",
    snomedCode: QCOVID_CODE_NONE,
  },
  'has-learning-disability-excluding-downs-syndrome': {
    label: "Has learning disability excluding Down's syndrome",
    snomedCode: 'LEARNING_DISABILITY',
  },
  'has-downs-syndrome': {
    label: "Has Down's syndrome",
    snomedCode: 'DOWNS_SYNDROME',
  },

  // chemotherapy
  'no-chemotherapy-in-the-last-12-months': {
    label: 'No chemotherapy in the last 12 months',
    snomedCode: QCOVID_CODE_NONE,
  },

  // chemotherapy: group A
  '5fu-single-agent': {
    label: '5FU single agent',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  abiraterone: { label: 'Abiraterone', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  anagrelide: { label: 'Anagrelide', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'aromatase-inhibitors': {
    label: 'Aromatase inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  'bevacizumab-single-agent': {
    label: 'Bevacizumab single agent',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  bisphosphonate: {
    label: 'Bisphosphonate',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  busulfan: { label: 'Busulfan ', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'capecitabine-single-agent': {
    label: 'Capecitabine single agent',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  'cdk4-6-inhibitors': {
    label: 'CDK4/6 inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  cetuximab: { label: 'Cetuximab', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'cisplatin-based-regimens': {
    label: 'Cisplatin based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  denosumab: { label: 'Denosumab ', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  durvalumab: { label: 'Durvalumab', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  enzalutamide: { label: 'Enzalutamide', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  fulvestrant: { label: 'Fulvestrant', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  hydroxycarbamide: {
    label: 'Hydroxycarbamide',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  'interferon-all-formulations': {
    label: 'Interferon (all formulations)',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  ipilimumab: { label: 'Ipilimumab', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  lenvatinib: { label: 'Lenvatinib', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  methotrexate: { label: 'Methotrexate', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'mitomycin-c': { label: 'Mitomycin C', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'mtor-inhibitors': {
    label: 'mTOR inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  nivolumab: {
    label: 'Nivolumab single agent',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  panitumumab: { label: 'Panitumumab', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'parp-inhibitors': {
    label: 'PARP inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  pembrolizumab: {
    label: 'Pembrolizumab single agent',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  pemetrexed: { label: 'Pemetrexed', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  raltitrexed: { label: 'Raltitrexed', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  regorafinib: { label: 'Regorafinib', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'single-agent-atezolizumab': {
    label: 'Atezolizumab single agent',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  sorafenib: { label: 'Sorafenib', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  tamoxifen: { label: 'Tamoxifen', snomedCode: 'SACT_CHEMOTHERAPY_A' },
  'taxane-weekly': {
    label: 'Taxane â€“ weekly',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  'trastuzumab-pertuzumab': {
    label: 'Trastuzumab +/- pertuzumab',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },
  'tyrosine-kinase-inhibitors-including-alk': {
    label: 'Tyrosine kinase inhibitors (including ALK &/or ROS)',
    snomedCode: 'SACT_CHEMOTHERAPY_A',
  },

  // chemotherapy: group B
  abvd: { label: 'ABVD', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'anthracycline-based-regimens': {
    label: 'Anthracycline based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  'asparaginase-based-regimens': {
    label: 'Asparaginase based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  beacopp: { label: 'BEACOPP', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  bendamustine: { label: 'Bendamustine', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'brentuximab-vedotin': {
    label: 'Brentuximab vedotin',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  'btk-inhibitors': {
    label: 'BTK inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  cabazitaxel: { label: 'Cabazitaxel', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'carboplatin-based-regimens': {
    label: 'Carboplatin based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  chop: { label: 'CHOP', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  chorambucil: { label: 'Chorambucil', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  cladrabine: { label: 'Cladrabine', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  cmf: { label: 'CMF', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  cvad: { label: 'CVAD', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'cyclophosphamide-fludarabine-combinations': {
    label: 'Cyclophosphamide/Fludarabine combinations',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  'dacarbazine-based-regimens': {
    label: 'Dacarbazine based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  daratumumab: { label: 'Daratumumab', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  dhap: { label: 'DHAP', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  eshap: { label: 'ESHAP', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'etoposide-based-regimens': {
    label: 'Etoposide based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  fec: { label: 'FEC', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  gdp: { label: 'GDP', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  gemcitabine: { label: 'Gemcitabine', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  ice: { label: 'ICE', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'ifophosphamide-based-regimens': {
    label: 'Ifophosphamide based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  imids: { label: 'IMIDs', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'irinotecan-and-oxaliplatin-based-regimens': {
    label: 'Irinotecan and Oxaliplatin based regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  'jak-inhibitors': {
    label: 'JAK inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  'liposomal-doxorubicin': {
    label: 'Liposomal doxorubicin',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  lomustine: { label: 'Lomustine', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  mogalizumab: { label: 'Mogalizumab', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  mvac: { label: 'MVAC', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'nab-paclitaxel': {
    label: 'Nab-paclitaxel',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  obinutuzumab: { label: 'Obinutuzumab', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  pentostatin: { label: 'Pentostatin', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'pi3kinase-inhibitors': {
    label: 'PI3Kinase inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  'proteosome-inhibitors': {
    label: 'Proteosome inhibitors',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  rituximab: { label: 'Rituximab', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'taxane-3-weekly': {
    label: 'Taxane â€“ 3 weekly',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  temozolomide: { label: 'Temozolomide', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  topotecan: { label: 'Topotecan', snomedCode: 'SACT_CHEMOTHERAPY_B' },
  'trastuzumab-emtansine': {
    label: 'Trastuzumab-emtansine',
    snomedCode: 'SACT_CHEMOTHERAPY_B',
  },
  ventoclax: { label: 'Ventoclax', snomedCode: 'SACT_CHEMOTHERAPY_B' },

  // chemotherapy: group C
  'all-aml-regimens': {
    label: 'All ALL / AML regimens',
    snomedCode: 'SACT_CHEMOTHERAPY_C',
  },
  bep: { label: 'BEP', snomedCode: 'SACT_CHEMOTHERAPY_C' },
  'highly-immunosuppressive-chemotherapy': {
    label:
      'Highly immunosuppressive chemotherapy (e.g. FluDAP, high dose Methotrexate & Cytarabine)',
    snomedCode: 'SACT_CHEMOTHERAPY_C',
  },
  'trifuradine-tipracil': {
    label: 'Trifuradine/Tipracil',
    snomedCode: 'SACT_CHEMOTHERAPY_C',
  },
};

export const getSnomedCodeForValue = (valueId) =>
  VALUE_OBSERVATIONS[valueId].snomedCode;

export const getLabelForSnomedCode = (snomedCode) => {
  if (snomedCode === SNOMED_CODE_NONE || snomedCode === QCOVID_CODE_NONE) {
    return 'None';
  }

  if (snomedCode === 'SACT_CHEMOTHERAPY_A') {
    return 'Chemotherapy systemic anti cancer therapy (SACT) protocol A';
  }

  if (snomedCode === 'SACT_CHEMOTHERAPY_B') {
    return 'Chemotherapy systemic anti cancer therapy (SACT) protocol B';
  }

  if (snomedCode === 'SACT_CHEMOTHERAPY_C') {
    return 'Chemotherapy systemic anti cancer therapy (SACT) protocol C';
  }

  try {
    const { label } = Object.values(VALUE_OBSERVATIONS).find(
      (value) => value.snomedCode === snomedCode
    );
    return label;
  } catch (e) {
    throw new Error(`cannot find label for: ${snomedCode} : ${e.message}`);
  }
};

const getLabelForValue = (valueId) => VALUE_OBSERVATIONS[valueId].label;

export const getObservationOption = (name) => ({
  label: getLabelForValue(name),
  name,
});
