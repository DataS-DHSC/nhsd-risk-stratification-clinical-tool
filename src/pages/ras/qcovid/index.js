import PropTypes from 'prop-types';
import HeadTags from '../../../components/HeadTags';
import Start from '../../../components/qcovid/Start';
import { ROUTE_QCOVID, ROUTE_PATIENT_DETAILS } from '../../../constants/routes';
import handleServerSideProps from '../../../utils/handleServerSideProps';

const Index = ({ prevData, featureFlags }) => {
  const d = new Date();
  d.setTime(d.getTime() + 12 * 24 * 60 * 60 * 1000);

  return (
    <>
      <HeadTags title="" description="" featureFlags={featureFlags} />
      <Start route={ROUTE_QCOVID + ROUTE_PATIENT_DETAILS} prevData={prevData} />
    </>
  );
};

Index.propTypes = {
  prevData: PropTypes.shape().isRequired,
  featureFlags: PropTypes.shape(),
};

Index.defaultProps = {
  featureFlags: {},
};

export const getServerSideProps = handleServerSideProps({ allowGet: true });

export default Index;
