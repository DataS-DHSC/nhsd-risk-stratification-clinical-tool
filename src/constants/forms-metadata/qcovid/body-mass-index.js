import { getLabelForField } from '../../../algorithm-mapping/qcovid/fields-mapping';
import { minMaxValidator } from '../../../utils/field_validators/minMaxValidator';

export const formId = 'body-mass-index';

export const BODY_MASS_INDEX_FORM = {
  label: "Calculate the patient's body mass index (BMI)",
  formId,
  sections: [
    {
      label: '',
      fields: {
        height: {
          label: getLabelForField('height'),
          ariaLabel: 'Height in centimetres',
          hint:
            'Without decimal places. The service can only give a result for a height from 140cm to 210cm.',
          hintAriaLabel:
            'Without decimal places. The service can only give a result for a height from 140 centimetres to 210 centimetres.',
          conversion: 'cm-to-ft-in',
          type: 'numeric-input',
          min: 140,
          max: 210,
          fieldValidator: (value) =>
            value ? minMaxValidator(value, 140, 210) : '',
          default: '',
        },
        weight: {
          label: getLabelForField('weight'),
          ariaLabel: 'Weight in kilograms',
          hint:
            'Without decimal places. The service can only give a result for a weight from 40kg to 180kg.',
          hintAriaLabel:
            'Without decimal places. The service can only give a result for a weight from 40 kilograms to 180 kilograms.',
          conversion: 'kg-to-st-lb',
          type: 'numeric-input',
          min: 40,
          max: 180,
          fieldValidator: (value) =>
            value ? minMaxValidator(value, 40, 180) : '',
          default: '',
        },
        // not part of the qCovid inputs
        'bmi-calc': {
          label: "The patient's BMI score is: ",
          default: 'Enter height and weight to calculate BMI.',
          type: 'bmi-calc',
          links: ['weight', 'height'],
        },
      },
    },
  ],
};
