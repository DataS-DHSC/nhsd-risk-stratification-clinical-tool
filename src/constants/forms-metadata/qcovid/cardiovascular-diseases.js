import { getLabelForField } from '../../../algorithm-mapping/qcovid/fields-mapping';

const formId = 'cardiovascular-diseases';

export const CARDIOVASCULAR_DISEASES_FORM = {
  label: 'Cardiovascular diseases',
  formId,
  sections: [
    {
      label: '',
      fields: {
        'has-atrial-fibrillation': {
          label: getLabelForField('has-atrial-fibrillation'),
          type: 'radio',
          default: 'false',
        },
        'has-coronary-heart-disease': {
          label: getLabelForField('has-coronary-heart-disease'),
          type: 'radio',
          default: 'false',
        },
        'has-heart-failure': {
          label: getLabelForField('has-heart-failure'),
          type: 'radio',
          default: 'false',
        },
        'has-congenital-heart-disease': {
          label: getLabelForField('has-congenital-heart-disease'),
          type: 'radio',
          default: 'false',
        },
        'has-peripheral-vascular-disease': {
          label: getLabelForField('has-peripheral-vascular-disease'),
          type: 'radio',
          default: 'false',
        },
      },
    },
  ],
};
