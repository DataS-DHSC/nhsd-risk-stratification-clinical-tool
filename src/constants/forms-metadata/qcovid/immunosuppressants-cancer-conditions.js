import { getLabelForField } from '../../../algorithm-mapping/qcovid/fields-mapping';
import { chemotherapyDrugOptions } from './chemotherapyDrugOptions';

export const formId = 'immunosuppressants-cancer-conditions';

export const IMMUNO_CANCER_CONDITIONS_FORM = {
  label: 'Immunosuppressants, cancer conditions & treatments',
  formId,
  sections: [
    {
      label: '',
      fields: {
        'lung-or-oral-cancer': {
          label: getLabelForField('lung-or-oral-cancer'),
          type: 'radio',
          default: 'false',
        },
        'solid-organ-transplant': {
          label: getLabelForField('solid-organ-transplant'),
          type: 'radio',
          default: 'false',
        },
        'cancer-of-the-blood-or-bone-marrow': {
          label: getLabelForField('cancer-of-the-blood-or-bone-marrow'),
          type: 'radio',
          default: 'false',
        },
        'bone-marrow-or-stem-cell-transplant': {
          label: getLabelForField('bone-marrow-or-stem-cell-transplant'),
          type: 'radio',
          default: 'false',
        },
        'had-radiotherapy-last-6-months': {
          label: getLabelForField('had-radiotherapy-last-6-months'),
          type: 'radio',
          default: 'false',
        },
        'chemotherapy-in-the-last-12-months': {
          label: getLabelForField('chemotherapy-in-the-last-12-months'),
          hint:
            "If they've had multiple chemotherapy agents, select the most immunosuppressive (of Group C first, then B, then A)",
          type: 'select',
          required: true,
          default: 'no-chemotherapy-in-the-last-12-months',
          options: chemotherapyDrugOptions(),
        },
        'prescribed-oral-prednisolone-steroids': {
          label: getLabelForField('prescribed-oral-prednisolone-steroids'),
          hint:
            'Prescribed by a clinician four or more times in the last 6 months.',
          type: 'radio',
          default: 'false',
        },
        'prescribed-immunosuppressants': {
          label: getLabelForField('prescribed-immunosuppressants'),
          hint:
            'Prescribed by a clinician four or more times in the last 6 months.',
          type: 'radio',
          default: 'false',
        },
      },
    },
  ],
};
