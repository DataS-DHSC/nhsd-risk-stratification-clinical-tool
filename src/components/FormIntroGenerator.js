import PropTypes from 'prop-types';
import { Hint, InsetText } from 'nhsuk-react-components';
import { ROUTE_CLINICAL_GUIDANCE } from '../constants/routes';

import { formId as qcovidBodyMassIndexFormId } from '../constants/forms-metadata/qcovid/body-mass-index';
import { formId as qcovidImunoCancerConditionsFormId } from '../constants/forms-metadata/qcovid/immunosuppressants-cancer-conditions';

// Adding new form intros?
// Think about extracting them into different files if this file is getting too long

export const BodyMassIndexFormIntro = () => (
  <>
    <InsetText>
      <p>
        BMI is an important and modifiable risk factor. You do not need to enter
        this information, but it&apos;ll help get a more accurate result if you
        do.
      </p>
    </InsetText>

    <Hint>If the patient&apos;s BMI is:</Hint>
    <Hint>
      <ul>
        <li>not entered, we&apos;ll use a BMI of 25</li>
        <li>below 15, we&apos;ll use a calculation of 15</li>
        <li>over 47, we&apos;ll use a calculation of 47</li>
      </ul>
    </Hint>
  </>
);

export const ImmunoCancerConditionsFormIntro = () => (
  <>
    <InsetText>
      <p>
        You can{' '}
        <a
          href={`${ROUTE_CLINICAL_GUIDANCE}#chemotherapy-and-immunosuppressants`}
          target="_blank"
          rel="noreferrer"
        >
          see a full list of immunosuppressants (opens in new window)
        </a>{' '}
        if you need guidance to answer the last question on this page.
      </p>
    </InsetText>
  </>
);

const FormIntroGenerator = ({ formId }) => {
  switch (formId) {
    case qcovidBodyMassIndexFormId:
      return <BodyMassIndexFormIntro />;
    case qcovidImunoCancerConditionsFormId:
      return <ImmunoCancerConditionsFormIntro />;
    default:
      return null;
  }
};

FormIntroGenerator.propTypes = {
  formId: PropTypes.string,
};

FormIntroGenerator.defaultProps = {
  formId: null,
};

export default FormIntroGenerator;
