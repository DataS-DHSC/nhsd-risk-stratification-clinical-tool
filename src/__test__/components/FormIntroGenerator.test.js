import { render } from '@testing-library/react';
import FormIntroGenerator from '../../components/FormIntroGenerator';

import { formId as qcovidBodyMassIndexFormId } from '../../constants/forms-metadata/qcovid/body-mass-index';
import { formId as qcovidImunoCancerConditionsFormId } from '../../constants/forms-metadata/qcovid/immunosuppressants-cancer-conditions';

const setup = (formId) => {
  const props = {
    formId,
  };

  return render(<FormIntroGenerator {...props} />);
};

describe('FormIntroGenerator', () => {
  it('should match snapshot for body mass index qcovid form intro type', () => {
    const container = setup(qcovidBodyMassIndexFormId);
    expect(container.asFragment()).toMatchSnapshot();
  });

  it('should match snapshot for immuno cancer conditions qcovid form intro type', () => {
    const container = setup(qcovidImunoCancerConditionsFormId);
    expect(container.asFragment()).toMatchSnapshot();
  });
});
