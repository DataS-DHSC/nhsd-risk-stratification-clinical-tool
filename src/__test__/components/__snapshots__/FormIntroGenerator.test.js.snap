// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormIntroGenerator should match snapshot for body mass index qcovid form intro type 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-inset-text"
  >
    <span
      class="nhsuk-u-visually-hidden"
    >
      Information: 
    </span>
    <p>
      BMI is an important and modifiable risk factor. You do not need to enter this information, but it'll help get a more accurate result if you do.
    </p>
  </div>
  <span
    class="nhsuk-hint"
  >
    If the patient's BMI is:
  </span>
  <span
    class="nhsuk-hint"
  >
    <ul>
      <li>
        not entered, we'll use a BMI of 25
      </li>
      <li>
        below 15, we'll use a calculation of 15
      </li>
      <li>
        over 47, we'll use a calculation of 47
      </li>
    </ul>
  </span>
</DocumentFragment>
`;

exports[`FormIntroGenerator should match snapshot for immuno cancer conditions qcovid form intro type 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-inset-text"
  >
    <span
      class="nhsuk-u-visually-hidden"
    >
      Information: 
    </span>
    <p>
      You can 
      <a
        href="https://digital.nhs.uk/coronavirus/risk-assessment/clinical-tool/guidance-for-clinicians#chemotherapy-and-immunosuppressants"
        rel="noreferrer"
        target="_blank"
      >
        see a full list of immunosuppressants (opens in new window)
      </a>
       if you need guidance to answer the last question on this page.
    </p>
  </div>
</DocumentFragment>
`;
